C/Makefile

```diff
@@ -63,8 +63,10 @@ build-avr-opt:$(AVR_OBJS)
 # make build-app path=path/to/app
 # TODO(@Mariamahmed44): #97 Implement this rule
 # ------------------------------------------
-build-app:
-	@echo "NOT IMPLEMENTED"
+build-app: $(AVR_OBJS)
+	@$(AVR_GCC) $(AVR_GCCFLAGS) $(AVR_OBJS) $(path)/*.c -o $(path)/$(AVR_ELF_TARGET)
```

Comment:
We have to consider case of running `make build-app` incorrectly for example: <br>
for example : <br>
    1] Running without specifying `src` `make build-app`